### ISSUE TITLE
Model fails to converge on transfer to audio backtesting problem

### ISSUE THREAD

**Issue by yangma12** [YuanGongND/ssast#19]
Dear Yuan and authors,
       First of all, thank you for your paper. Recently, I migrated your pre-trained model to the regression prediction task of personality computing. After splicing several fully connected layers after your original model, the result is that the predicted value will only be maintained at a very low level during training. In a small interval, there will be no effective changes. Have you done relevant regression experiments? What are the possible reasons for this problem?
        Sorry to bother you with my question and thank you very much for reading my question

yang

**Comment by YuanGongND** [comment]
hi there, 

Do you mean you finetune our pretrained model for a regression task? 

What do you by this?

> After splicing several fully connected layers after your original model

-Yuan

**Comment by yangma12** [comment]
thank you for your reply！I mainly use this data set for fine-tuning, and separate the audio of this data set（https://chalearnlap.cvc.uab.cat/dataset/24/description/）. Each audio is a 15-second speech audio, and the MLP is stitched after the model to adjust the dimension of the audio data output by the final model to ( batchsize,5), 5 corresponds to the regression value of five personality traits corresponding to an audio.

**Comment by yangma12** [comment]
> 

In the experiment, I tried to adjust the learning rate and other parameters, tried to remove the mask and mixing in the data preprocessing, set the input_tdim to 1530 to suit my audio length, label_dim to 512, and finally performed regression prediction through the following code : nn.Sequential(
            nn.Linear(in_features=512, out_features=256),
            nn.ReLU(inplace=True),
            nn.Linear(in_features=256, out_features=128),
            nn.ReLU(inplace=True),
            nn.Linear(in_features=128, out_features=6),
            nn. Sigmoid()
        )，Forgive me for not being deep enough in deep learning at the moment, I'm not sure where the problem might be.

**Comment by YuanGongND** [comment]
There are a few things:

1. First, it seems a multi-modal, speech-dominated dataset. So you might want to try an audio-visual model or speech-based model (e.g., Hubert), according to my experience, for pure speech task, pure speech models are better, can you see the Table 5 of [SSAST Paper](https://ojs.aaai.org/index.php/AAAI/article/download/21315/21064)? For audio-visual models, we have [CAV-MAE](https://openreview.net/pdf?id=QPtMRyk5rb) for general audio-visual model, but again, you might need a model focusing on face. 

2. For this

> nn.Sequential(
nn.Linear(in_features=512, out_features=256),
nn.ReLU(inplace=True),
nn.Linear(in_features=256, out_features=128),
nn.ReLU(inplace=True),
nn.Linear(in_features=128, out_features=6),
nn. Sigmoid()
)

Is Sigmoid common for regression? Setting "label_dim to 512" (for classification) and then a few dense layers seems to be redundent. You can just change the last MLP layer to a regression head.

https://github.com/YuanGongND/ssast/blob/a1a3eecb94731e226308a6812f2fbf268d789caf/src/models/ast_models.py#L166-L167

But I know very little about your task. You need to tune the params by yourself. For some networks, we use a larger learning rate for the mlp layer because it is random initialized while other parameters are pretrained.

I mainly answer questions that are related to what we presented in the paper, and it is hard for me to answer questions regarding new task / usage of the model. 

-Yuan

**Comment by YuanGongND** [comment]
Another minor point is that you said there are 5 regression values, but `nn.Linear(in_features=128, out_features=6)` shows 6.

**Comment by yangma12** [comment]
> There are a few things:
> 
> 1. First, it seems a multi-modal, speech-dominated dataset. So you might want to try an audio-visual model or speech-based model (e.g., Hubert), according to my experience, for pure speech task, pure speech models are better, can you see the Table 5 of [SSAST Paper](https://ojs.aaai.org/index.php/AAAI/article/download/21315/21064)? For audio-visual models, we have [CAV-MAE](https://openreview.net/pdf?id=QPtMRyk5rb) for general audio-visual model, but again, you might need a model focusing on face.
> 2. For this
> 
> > nn.Sequential(
> > nn.Linear(in_features=512, out_features=256),
> > nn.ReLU(inplace=True),
> > nn.Linear(in_features=256, out_features=128),
> > nn.ReLU(inplace=True),
> > nn.Linear(in_features=128, out_features=6),
> > nn. Sigmoid()
> > )
> 
> Is Sigmoid common for regression? Setting "label_dim to 512" (for classification) and then a few dense layers seems to be redundent. You can just change the last MLP layer to a regression head.
> 
> https://github.com/YuanGongND/ssast/blob/a1a3eecb94731e226308a6812f2fbf268d789caf/src/models/ast_models.py#L166-L167
> 
> But I know very little about your task. You need to tune the params by yourself. For some networks, we use a larger learning rate for the mlp layer because it is random initialized while other parameters are pretrained.
> 
> I mainly answer questions that are related to what we presented in the paper, and it is hard for me to answer questions regarding new task / usage of the model.
> 
> -Yuan

Thank you for your kind reply, thank you very much
1. This data set is a multi-modal task, but I try to do a single-modal case first. In the two modes of image and text, I use the pre-trained transformer model respectively to have good results, so Now I am mainly looking for a solution to the audio mode. 
2. When I first used resnet as the backbone network experiment, I found that adding sigmod would have a better effect, so I will continue to use it here. I tried to remove the back according to your suggestion today. A large number of mlp network layers, but still the same problem as before.
3.I did another experiment. On the ks2 task, the result of running on the test data set without modification is: Acc=0.9068, AUC=0.9974. Then I made a modification, repeating each data in the time dimension, that is, fbank.repeat(15,1), and then modified the input_tdim parameter of the model, and then the result from the test set is: Acc=0.3168 , AUC=0.9012. I think the length of the audio data will have a greater impact on the model?

**Comment by yangma12** [comment]
> 另一个小问题是你说有 5 个归因值，但`nn.Linear(in_features=128, out_features=6)`显示为 6。

This is because in addition to the indicators of personality characteristics, this data set also has an indicator of interview scoring, which is put together for experiments here, sorry forgot to explain
